frontend/src/lib/otel/shared.ts

```diff
@@ -86,8 +83,32 @@ export const traceHeaders = (span: Span, type: 'request' | 'response', headers:
   });
 };
 
+function getUser(event: RequestEvent | NavigationEvent | Event): LexAuthUser | null {
+  if (isBrowserEvent(event)) {
```

Comment:
Had to do it this way because of the way the Typescript checks work, but I can write different checks. (`ifBrowserEvent`, if it returns false, removes `Event` from the list of possibilities, at which point we are able to call `isNavigationEvent`. But I can change the types of the functions to achieve the same result.

Reply:
Oh, gotcha! Make isNavigationEvent should be more forgiving/also accept Events.

Reply:
Tweaked the Typescript types that `isRequestEvent` accepts just slightly, and thereby enabled myself to merge the two else clauses together. Now I'm much happier with it, and the other use of `isRequestEvent` is not affected.